timemachine/cpp/src/context.cu

```diff
@@ -179,7 +188,7 @@ void Context::_step(double lambda, unsigned long long *du_dl_out) {
             d_du_dx_t_, // we only need the forces
             nullptr,
             du_dl_out ? d_du_dl_buffer_ : nullptr,
-            nullptr,
+            compute_u ? d_u_buffer_ : nullptr,
```

Comment:
If the barostat will attempt a move, compute the energy immediately and pass it to the barostat.